docs/adr/0015-cluster-resources-amendment.md

```diff
@@ -0,0 +1,35 @@
+# Declarative Definitions for Cluster Scoped Resources
+
+- Status: accepted
+- Deciders: larsks, HumairAK, tumido
+
+Technical Story: [issue-1][1]
+
+## Context and Problem Statement
+
+In [ADR-0015][2] we laid out a directory structure of organizing cluster scoped resources. For the most part this has worked out relatively well. However, one thing that was not factored into this structure were the possibilities of name conflicts. Consider the following scenario:
+
+Suppose we have:
+1) A resource of kind Project named example-project with an apiversion: project.openshift.io/v1
+2) A resource of kind Project named example-project with an apiversion: config.openshift.io/v1
+
+Note that their API group differs, but everything else is the same. Both of these resources can exist simultaneously on a live cluster. However, these cannot co-exist in our github repo at `operate-first/apps/cluster-scope`, if we are to use the structuring as defined in [ADR-0015][2]. This is because, using the current structure, they would both need to be stored under `operate-first/apps/cluster-scope/base/projects/example-project/project.yaml`.
+
+
+## Considered Options
+
+The general consensus has been to keep the current structure and amend it to factor in the API Group. Some suggestions are:
+
+1. Use the following schema: `cluster-scope/base/<API_GROUP>-<KIND_PLURAL>/<METADATA_NAME>/<KIND_SINGULAR>.yaml`
+  This would factor in the API group as part of the name, and would not require additional nesting of directories
+2. Use the following schema: `cluster-scoped-resources/base/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml`
+  This would require additional nesting of directories, but this method has been observed in other tooling (i.e. how must-gather does their organization)
+
+## Preferred option
+
+We will opt for the second option. The first option may introduce yet another custom way to organize manifests, while the second one seems like a tried and tested method. In the case of the core API group (i.e. where resources like [namespaces][3] belong), they will be stored at the `cluster-scope/base` path (e.g. `cluster/scope/base/namespaces/...`).
```

Comment:
I'd like to vote for putting `core` resource types in `cluster-scope/base/core/`, because that makes a bunch of things easier: e.g., `base/*/*/*yaml` gets you all the files, there's less special casing when writing automation, etc.

Reply:
that's a fair point, I'm okay with this @tumido ?

Reply:
I don't see much of an issue with that. Like.. does it make any difference if you query for the files as `base/**/*.yaml` ? 

I'd be against inventing new api group here. The api group in the manifest is exactly `None`, `Null`, not `Core`. I think this makes it easier to parse the resource when categorizing it - `apiVersion` property can be split on `/` and you just use the first portion of it directly as your folder name.

Reply:
I'd rather see us be consistent in our filesystem layout. I really think that having core resources set at a different level in the hierarchy from all other resources is just going to complicate things.

Reply:
> I wouldn't say we're inventing a new core group if k8s docs and openshift api ref both also call it thus?

Hm, they are calling it this way, but the API is not reflecting this name anywhere and the apiVersion and group either. What the kubectl/oc client basically does is that it takes the `apiVersion` from manifests as is and uses it as an API path. There's no "core" name involved.

I'm still have some reservations on the simplicity of automatic parsing on the resources, because we now need to map certain resources to a new name that is not present in the `apiVersion` string. But sure, let's try it out with the `core` folder than...

Reply:
Okay let's go with `core` then, we can always revisit if it does indeed give us issues.